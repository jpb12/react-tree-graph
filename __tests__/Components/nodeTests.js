import React from 'react';
import { shallow } from 'enzyme';

import Node from '../../src/components/node';

const defaultProps = {
	x: 1,
	y: 2,
	keyProp: '',
	labelProp: 'name',
	gProps: {
		className: 'test'
	},
	circleProps: {},
	textProps: {},
	offset: 3,
	radius: 4,
	name: 'Test Node'
};

describe('<Node>', () => {
	test('renders correctly', () => {
		const tree = shallow(<Node {...defaultProps}/>);
		expect(tree).toMatchSnapshot();
	});

	test('click event has correct parameters', () => {
		const clickMock = jest.fn();
		const event = {};

		const props = {
			keyProp: 'id',
			gProps: {
				onClick: clickMock
			},
			id: 'testKey'
		};

		const tree = shallow(<Node {...defaultProps} {...props}/>);
		tree.find('g').simulate('click', event);

		expect(clickMock).toHaveBeenCalledTimes(1);
		expect(clickMock).toHaveBeenCalledWith(event, 'testKey');
	});

	test('right click event has correct parameters', () => {
		const rightClickMock = jest.fn();
		const event = {};

		const props = {
			keyProp: 'id',
			gProps: {
				onContextMenu: rightClickMock
			},
			id: 'testKey'
		};

		const tree = shallow(<Node {...defaultProps} {...props}/>);
		tree.find('g').simulate('contextmenu', event);

		expect(rightClickMock).toHaveBeenCalledTimes(1);
		expect(rightClickMock).toHaveBeenCalledWith(event, 'testKey');
	});


	test('clicking with no prop handler does nothing', () => {
		const props = {
			keyProp: 'id',
			id: 'testKey'
		};

		const tree = shallow(<Node {...defaultProps} {...props}/>);
		tree.find('g').simulate('click');
	});

	test('htmlProps applied to all elements', () => {
		const props = {
			gProps: {
				className: 'g'
			},
			circleProps: {
				className: 'circle'
			},
			textProps: {
				className: 'text'
			}
		};

		const tree = shallow(<Node {...defaultProps} {...props}/>);
		expect(tree).toMatchSnapshot();
	});
});