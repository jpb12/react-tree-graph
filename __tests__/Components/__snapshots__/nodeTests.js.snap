// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Node> htmlProps applied to all elements 1`] = `
<g
  className="g"
  transform="translate(2, 1)"
>
  <circle
    className="circle"
    r={4}
  />
  <text
    className="text"
    dx={4.5}
    dy={3}
  >
    Test Node
  </text>
</g>
`;

exports[`<Node> renders correctly 1`] = `
<g
  className="test"
  transform="translate(2, 1)"
>
  <circle
    r={4}
  />
  <text
    dx={4.5}
    dy={3}
  >
    Test Node
  </text>
</g>
`;
